### PR TITLE
[posix] forkpty-arg value as a single argument

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: src/posix/platform/system.cpp [hunk 1]
```diff
@@ -52,7 +52,16 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
...
 
 #if OPENTHREAD_POSIX_VIRTUAL_TIME
-    virtualTimeInit(static_cast<uint16_t>(atoi(radioUrl.GetValue("forkpty-arg"))));
+    // The last argument must be the node id
+    {
+        const char *nodeId = radioUrl.GetValue("forkpty-arg");
+
+        for (const char *arg = nodeId; (arg = radioUrl.GetValue("forkpty-arg", nodeId)) != nullptr; nodeId = arg)
+        {
+        }
```
Comment (simonlingoogle) [inline src/posix/platform/system.cpp:61]:
```suggestion
        const char *nodeId = radioUrl.GetValue("forkpty-arg");
        const char *arg;

        while ((arg = radioUrl.GetValue("forkpty-arg", nodeId)) != nullptr)
        {
            nodeId = arg;
        }
```

Assigning `arg = nodeId` is useless?

File: src/posix/platform/hdlc_interface.cpp [hunk 2]
```diff
@@ -599,38 +595,40 @@ int HdlcInterface::ForkPty(const char *aCommand, const char *aRadioUrl)
...
-        char      cmd[kMaxCommand];
+        constexpr int kMaxArguments = 32;
+        char *        argv[kMaxArguments];
+        size_t        index = 0;
 
-        if (aRadioUrl == nullptr)
+        argv[index++] = const_cast<char *>(aRadioUrl.GetPath());
+
+        for (const char *arg = nullptr;
+             index < OT_ARRAY_LENGTH(argv) && (arg = aRadioUrl.GetValue("forkpty-arg", arg)) != nullptr;
```
Comment (simonlingoogle) [inline src/posix/platform/hdlc_interface.cpp:610]: Die if user specify more than 32 arguments?

File: src/posix/platform/hdlc_interface.cpp [hunk 3]
```diff
@@ -599,38 +595,40 @@ int HdlcInterface::ForkPty(const char *aCommand, const char *aRadioUrl)
...
-        VerifyOrExit((pid = forkpty(&fd, nullptr, &tios, nullptr)) != -1, perror("forkpty()"));
+        VerifyOrDie((pid = forkpty(&fd, nullptr, &tios, nullptr)) != -1, OT_EXIT_ERROR_ERRNO);
     }
 
     if (0 == pid)
     {
-        const int kMaxCommand = 255;
-        char      cmd[kMaxCommand];
+        constexpr int kMaxArguments = 32;
+        char *        argv[kMaxArguments];
```
Comment (simonlingoogle) [inline src/posix/platform/hdlc_interface.cpp:604]: nit:
```suggestion
        char *        argv[kMaxArguments+1];
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

File: src/posix/platform/system.cpp (hunk 1 above)
Comment (bukepo) [reply src/posix/platform/system.cpp:61]:
```suggestion
        const char *nodeId = nullptr;

        for (const char *arg = nullptr; (arg = radioUrl.GetValue("forkpty-arg", arg)) != nullptr; nodeId = arg)
        {
        }
```

File: src/posix/platform/hdlc_interface.cpp (hunk 3 above)
Comment (bukepo) [reply src/posix/platform/hdlc_interface.cpp:604]:
```suggestion
        char *        argv[kMaxArguments + 1];
```

**Comment by jwhui** [comment]
> [posix] url support space in forkpty-arg

Is this PR title accurate?

**Comment by bukepo** [comment]
> > [posix] url support space in forkpty-arg
> 
> Is this PR title accurate?

updated.

**Review by jwhui — APPROVED** [review]

**Pull request merged**